### PR TITLE
Adds pivotalR wrapper for MADlib kmeans clustering functions.

### PR DESCRIPTION
**Pull request by orhankislal**
Please note that this wrapper requires the following commit for the MADlib to work: https://github.com/apache/incubator-madlib/pull/68/commits/d3e87b514112b324a9475b60f50a590e651f611c

### CODE REVIEW

**Comment by orhankislal** [comment]
@iyerr3 The output class is the actual kmeans from the stats package, which means we are using the exactly same functions. Transition between R kmeans and madlib.kmeans should be quite easy. I can create a separate output class with its own functions but I am not sure if that will improve the usability.

**Comment by iyerr3** [comment]
@orhankislal Got it.
Now help me understand why we even need the `setClass("kmeans.madlib")` if
we're not using that class.

**Comment by orhankislal** [comment]
@iyerr3 Now I see the source of confusion. It is not necessary, just a leftover line from an earlier stage of development. I'll push a commit to remove it.

File: man/madlib.kmeans.Rd [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  }
+
+  \item{centers}{
+
+    A number, a matrix or a string which indicates how the centers are
+    formed by the function. If it is a number, this sets the number of
+    target centroids and the random (or kmeans++) seeding method is
+    used. Otherwise, this parameters is used for initial centers. If
+    it is a matrix, its rows will denote the initial centroid
+    coordinates. Otherwise this parameter will contain the name of a
```
Comment (iyerr3) [inline man/madlib.kmeans.Rd]: Can the `centers` also be a `db.data.frame` that points to a table?

File: man/madlib.kmeans.Rd [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  }
+  \item{algorithm}{
+    The algorithm to compute the kmeans. Currently disabled (default:
+    ``\code{Lloyd}'') and kept for the future implementations.
+  }
+  \item{pid}{
+    Name of the column (from the table that is pointed by \code{x}) that
+    contains the ids for each point.
+  }
+  \item{expr}{
```
Comment (iyerr3) [inline man/madlib.kmeans.Rd]: The regular `kmeans` expects all columns of input data frame to be part of the clustering. Can we make that same assumption and build the array of features from all the columns?

File: man/madlib.kmeans.Rd [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    Name of the column (from the table that is pointed by \code{x}) that
+    contains the coordinate arrays for each point.
+  }
+  \item{expr.centroid}{
```
Comment (iyerr3) [inline man/madlib.kmeans.Rd]: If we allow input of db.data.frame then we can avoid passing this character, isn't it?

File: man/madlib.kmeans.Rd (hunk 1 above)
Comment (orhankislal) [inline man/madlib.kmeans.Rd]: Yes, we can add this option as well.

Comment (iyerr3) [inline man/madlib.kmeans.Rd]: Now that I think about, IMO we shouldn't support the database name input but force the user to create a db.table even if it's done during the function call.

File: man/madlib.kmeans.Rd (hunk 2 above)
Comment (orhankislal) [inline man/madlib.kmeans.Rd]: The MADlib kmeans function does not require the `pid` which means we should only need a single column (that has the point arrays). However we do need the `pid` values for the ordering of `closest column` results (presented as `clustering vector').

Comment (iyerr3) [inline man/madlib.kmeans.Rd]: I'm not arguing against the `pid` but against the `expr` input. We can assume that all columns given by the user will be used for clustering and build an array out of that.

File: man/madlib.kmeans.Rd (hunk 3 above)
Comment (orhankislal) [inline man/madlib.kmeans.Rd]: If we make the assumption that the initial centroids table has a single column then yes.

File: man/madlib.kmeans.Rd (hunk 2 above)
Comment (orhankislal) [inline man/madlib.kmeans.Rd]: `expr` is a single column that contains an array in each row. That is how MADlib kmeans expect the input. We can implement a MADlib kmeans that operates on a number of columns and add a wrapper for that.

File: man/madlib.kmeans.Rd (hunk 3 above)
Comment (iyerr3) [inline man/madlib.kmeans.Rd]: Why expose the database constructs to the R user? We can accept a `db.table` where the user can project data by passing just the appropriate column (eg. `init_centroid[col]`)

File: man/madlib.kmeans.Rd (hunk 2 above)
Comment (iyerr3) [inline man/madlib.kmeans.Rd]: My point is that the R input need not be the same as the SQL input. We accept a `db.table` where user has filtered the necessary columns. Within the R code we create a view (or temporary table) that has necessary `expr` column and then call the SQL function. This way R user is not responsible for creating such a column.

Comment (orhankislal) [inline man/madlib.kmeans.Rd]: I understand the point and I think that a generic function that allows R users to aggregate columns into an array will be useful for many MADlib functions. However, I thought the reason these arrays are used is the column number limit in PSQL systems. How should we handle the case where the user has a large dataset that is already aggregated into arrays?

File: man/madlib.kmeans.Rd (hunk 3 above)
Comment (orhankislal) [inline man/madlib.kmeans.Rd]: What if the user already has the centers in a table somewhere in the dataset?

**Pull request closed without merging**